Image api, first draft

zzyyfff opened the pull request:
+ Create image model
+ Add promise-ified S3 upload function 
+ Register 'image' routes in server.js 
+ Add 'image' CRUD routes
+ Add reference to images in user model 

All of these are UNTESTED but are being PR'd so that everyone can test them on development branch.

@zzyyfff + @juliojesse25

zzyyfff committed:
Create image model

+ Set up MVP schema for `image` resource in app/models/image.js
+ Resolves issue #3
+ Collaboration of @zzyyfff and @juliojesse25

zzyyfff committed:
Add promise-ified S3 upload function

+ Issue: #2
+ @zzyyfff and @juliojesse25

zzyyfff committed:
Register 'image' routes in server.js

+ Issue: #5
+ @zzyyfff + @juliojesse25

zzyyfff committed:
Add 'image' CRUD routes

+ Issues: #4, #6, #8, #10
+ Possibly issue #7, but needs coordination w/ frontend team
+ @zzyyfff + @juliojesse25

zzyyfff committed:
Add reference to images in user model

+ Issue: #9
+ @zzyyfff + juliojesse25

rogu2 reviewed (changes requested):
Looks great! We need to include npm packages for package-json:

multer
aws-sdk

zzyyfff committed:
Installed missing packages

+ installed aws-sdk
+ installed multer
+ resolves #12
+ @zzyyfff + @juliojesse25

zzyyfff commented:
Created issue #12 to address this

zzyyfff commented:
Installed missing packages, resolving #12 

Request another review: @rogu2 @asangurima
